code/workspaces/infrastructure-api/src/stacks/stackBuilders.js

```diff
@@ -28,6 +28,18 @@ export const createService = (params, generator) => () => {
     });
 };
 
+export const createHeadlessService = (params, generator) => () => {
+  const { name, projectKey, type } = params;
+  const rootServiceName = `${type}-${name}`;
+  const headlessServiceName = getHeadlessServiceName(rootServiceName);
```

Comment:
I don't think the Kubernetes API should be responsible for name generation. That is probably the responsibility of this file or the stack builder.